spynnaker/pyNN/models/neuron/population_machine_vertex.py

```diff
@@ -627,3 +628,35 @@ def gen_on_machine(self):
     def read_generated_connection_holders(self, transceiver, placement):
         self._app_vertex.synapse_manager.read_generated_connection_holders(
             transceiver, placement)
+
+    @overrides(AbstractReadParametersBeforeSet.read_parameters_from_machine)
+    def read_parameters_from_machine(
+            self, transceiver, placement, vertex_slice):
+
+        # locate SDRAM address to where the neuron parameters are stored
+        neuron_region_sdram_address = self.neuron_region_sdram_address(
+            placement, transceiver)
+
+        # shift past the extra stuff before neuron parameters that we don't
+        # need to read
+        neuron_parameters_sdram_address = (
+            neuron_region_sdram_address +
+            self._app_vertex.tdma_sdram_size_in_bytes +
+            self._app_vertex.BYTES_TILL_START_OF_GLOBAL_PARAMETERS)
```

Comment:
It seems odd to go to the trouble of moving this here, only to ask the app vertex anyway!  I have already suggested moving one of these properties above; would it make sense to move the other as well?

Reply:
yet again, the neuron impl is a data holder and so can live in the app vertex, asking the app vertex for data is considered perfectly fine. moving the BYTES_TILL_START_OF_GLOBAL_PARAMETERS doesnt on its own help, as the neuron impl is the one enforcing the tdma now,and as the app vertex is the one impling that api right now, you cant push the call for tdma size as its directly related to the app vertex.

Reply:
OK